source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -132,6 +143,11 @@ void EnvoyQuicClientStream::switchStreamBlockState() {
 
 void EnvoyQuicClientStream::OnInitialHeadersComplete(bool fin, size_t frame_len,
                                                      const quic::QuicHeaderList& header_list) {
+  if (stream_bytes_read() > bytesMeter()->wireBytesReceived()) {
```

Comment:
This isn't really counting header bytes, is it? From reading the code, it seems like stream_bytes_read() counts the total bytes flow into the stream sequencer, and we are sync'ing bytes meter with this counter whenever we can. Can it be wrapped into a helper function with a proper name? 

And another thought is that QuicStream::OnStreamFrame() is a virtual method, can it be overridden and update the byte meter there? Or extends the BytesMeter interfaces, i.e. overriding wireBytesReceived() to read QuicStream::stream_bytes_read() directly?